### ISSUE TITLE
Possibility to rename operators ?

### ISSUE THREAD

**Issue by J3m5** [crcn/sift.js#220]
Hi,
is there a way to rename operators ?

If not, I was thinking about omitting built in operators and then adding custom ones that would be the default one but with other names,
would it be doable ?

Thanks

**Comment by crcn** [comment]
Yep, this is doable: 

```javascript
import { createQueryTester, $eq, $in } from "sift";
const filter = createQueryTester({ $eq: 5 }, { operations: { $eq, $in } });
```

**Comment by crcn** [comment]
Let me know if this answers your question!

**Comment by J3m5** [comment]
Thanks,
I tried something like this
```js
const { createQueryTester, $eq } = require("sift");
const filter = createQueryTester({ $equal: 5 }, { operations: { $equal: $eq } });

const result = [1, 2, 3, 4, 5].filter(filter);

console.log(result); // [ 5 ]
```

So it works but my usecase was about removing the $ sign but I noticed that it doesn't works without.
Is there a way to have operators without the dollar sign ?

**Comment by crcn** [comment]
> So it works but my usecase was about removing the $ sign but I noticed that it doesn't works without.

Ahh, that's actually how Sift identifies operator keys - there isn't an easy way around that. What are you trying to do exactly?

**Comment by J3m5** [comment]
I'm trying to adapt my local queries for an api backend ([api-platform](https://github.com/api-platform/api-platform)).
It doesn't use the mongo syntax for it's queries (unlike [feathers.js](https://github.com/feathersjs/feathers)) so I wanted to 
directly change operator names instead of having to convert them on the fly.

The main goal here is to have the same syntax for querying the local store on the front-end and the api on the backend, like with [Feathers-vuex](https://github.com/feathersjs-ecosystem/feathers-vuex).

**Comment by marshallswain** [comment]
@J3m5 Have you tried adding an array of built-in key names to core.js and changing this line https://github.com/crcn/sift.js/blob/master/src/core.ts#L414 to check if the value is in that array.  Then you'd have to add custom keys to the same array.  I haven't looked at what it would take for the custom keys, yet.

**Comment by J3m5** [comment]
Hey @marshallswain, good point.

I saw this part of the code, but unfortunately it's not customizable on the user side.

But yes, checking if it's part of an existing set of operators that could be customized would be good enough for that.
It could even works faster with a hashmap instead of an array.

But I don't remember if the operators names are used elsewhere, this would need to be taken care of.

Would you accept a PR for that feature @crcn ?
If it's ok for you, I wouldn't mind some guidance.

**Comment by J3m5** [comment]
Or maybe it would be easier to do it with ucast https://www.github.com/stalniy/ucast/tree/master/packages%2Fjs%2FREADME.md

But we'd need to be able to replace sift in the setup.

**Comment by crcn** [comment]
@J3m5 oh boy I'm a month late on this, sorry about that. 🙇 

Here's a fix: https://github.com/crcn/sift.js/pull/229


Let me know if this addresses your issue!

**Comment by J3m5** [comment]
No problem, I wasn't even expecting you to do it.
Thanks @crcn , I've just tested it and it works great!

**Comment by crcn** [comment]
Awesome, glad it worked out! 👏 

This is published to NPM as `14.0.2`. Feel free to close out this issue.
